### PR TITLE
Merge: from master to dev [fix bug:注销接口地址造成死循环]

### PR DESCRIPTION
**Pull request by fangyincheng**
**What this PR does**:

  merge master

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: remoting/zookeeper/client.go [hunk 1]
```diff
@@ -299,29 +299,26 @@ func (z *ZookeeperClient) UnregisterEvent(zkPath string, event *chan struct{}) {
...
+	defer z.Unlock()
+	infoList, ok := z.eventRegistry[zkPath]
+	if !ok {
+		return
+	}
+	for i, e := range infoList {
+		if e == event {
+			arr := infoList
+			infoList = append(arr[:i], arr[i+1:]...)
+			logger.Debugf("zkClient{%s} unregister event{path:%s, event:%p}", z.name, zkPath, event)
```
Comment (AlexStocks) [inline remoting/zookeeper/client.go]: maybe we should adjust the log level to logger.Infof？

File: remoting/zookeeper/client.go [hunk 2]
```diff
@@ -299,29 +299,26 @@ func (z *ZookeeperClient) UnregisterEvent(zkPath string, event *chan struct{}) {
...
 		}
 	}
-	z.Unlock()
+	logger.Debugf("after zkClient{%s} unregister event{path:%s, event:%p}, array length %d",
```
Comment (AlexStocks) [inline remoting/zookeeper/client.go]: logger.Infof

File: remoting/zookeeper/client.go (hunk 1 above)
Comment (fangyincheng) [reply remoting/zookeeper/client.go]: `Info` is a higher log levle. I think `Info` should not print details, `Info` should be a key infomation.

Comment (fangyincheng) [reply remoting/zookeeper/client.go]: done

**Pull request merged**